src/libstore/filetransfer.cc

```diff
@@ -611,6 +744,12 @@ struct curlFileTransfer : public FileTransfer
 
     std::thread workerThread;
 
+public:
+    bool hasQuit()
+    {
+        return state_.lock()->quit;
+    }
```

Comment:
What's this?

Reply:
I see it has only one caller, where it was inlined before. I think it is not worth having this function

Reply:
I did it because `state_` is private

Reply:
Oh nevermind, I see what is going on now.